Add Support for Anchor Tagging to Differentiator #340

efuenzalida52 opened the pull request:
added anchor support for differentiators-section.

efuenzalida52 committed:
Add Support for Anchor Tagging #340

added anchor support for differentiators-section.

TaylorYeeBC commented on src/blocks/differentiator-section/block.json:
```diff
@@ -24,7 +24,14 @@
+	},
+	"supports": {
+		"anchor": true,
+		"customClassName": true,
```
Is this supposed to be true?

taija replied:
I'm also curious about this- why was this added as an option?

TaylorYeeBC commented on src/blocks/differentiator-section/render.php:
```diff
@@ -10,13 +10,38 @@
+//so Twig can add an `id` only if WP didn’t
+$context['wrapper_has_id'] = (strpos($wrapper, ' id=') !== false);
+
+/* // Building a stable heading id for aria-labelledby (optional)
```
Does this need to still be here?

TaylorYeeBC reviewed (approved):
Code Reviewed and tested on Staging site.
Left comments on code I had questions about, but current functionality is approved.

efuenzalida52 committed:
Add Support for Anchor Tagging #340

updated to hide additional css advanced block field

taija commented on src/blocks/differentiator-section/index.js:
```diff
@@ -16,8 +17,8 @@ const BLOCK_TEMPLATE = [
-registerBlockType( 'bc-sitka-spruce/differentiator-section', {
-
+registerBlockType( metadata.name, {
+	...metadata, // pulls in supports/attributes so anchor persists
```
Woah - this is super cool! Is this new? Love not needing to define the block name twice.

efuenzalida52 committed:
Add Support for Anchor Tagging #340

updated render to remove unneed code

taija commented on src/blocks/differentiator-section/style.scss:
```diff
@@ -41,7 +41,12 @@
+	 // (optional: also cover if the ID ends up on an inner element)
+	[id] { scroll-margin-top: var(--anchor-offset, 96px); }
 	margin-top: 0;
+	padding-top: 0ch;
```
Curious why this is needed?

efuenzalida52 replied:
I ran into an issue where the anchor ID was set but the frontend wasn't scrolling- a potential suggetsion/solution I found was that the shape/curved arch could be overlapping- this is supposed to keep the section from getting stuck tucked behind the arch .

padding-top is indded an "oops!" fixing...

taija commented on stories/differentiators/differentiators.twig:
The changes here are causing issues. 

1. The arch shape went away in my test environment
2. I'm overall concerned about the changes to how things are structured, as it means that this element will have a completely different structure than any other element that has an arch shape; this is bad for consistancy and maintainability
3. The story in storybook is no longer rendering correctly - something in this change broke it

TaylorYeeBC reviewed (changes requested):
Spoke too soon! Will approve after addressing Taija's findings.

efuenzalida52 committed:
Updating to improve structural changes to match & fix storybook.

efuenzalida52 committed:
Add Support for Anchor Tagging #340

Changed structure to better match current structure. Also fixed arch shape & storybook break

taija commented:
@efuenzalida52  - NPM is doing the annoying build failure it was doing to me a while ago in ADO. I'm going to pull this branch and take a look.

taija committed:
Merge branch 'release-v1.4' into feat-differentiator-anchor

Bring in changes from release branch to fix build issues in ADO

taija committed:
🐛 FIX: Attempt to Resolve NPM issue

Do a clean rebuild of package-lock.json

taija commented on stories/differentiators/differentiators.twig:
Minor fix- indentation is kinda messed up in this file now.

taija commented on stories/differentiators/differentiators.twig:
There is an issue here that Chromatic found:

## Pre-changes: all the headings align with eachother

<img width="1264" height="1226" alt="Image" src="https://github.com/user-attachments/assets/4c0eec42-03d2-4b46-a677-5f5d01cae648" />

## Post-changes: they no longer align

<img width="1309" height="1172" alt="Image" src="https://github.com/user-attachments/assets/4100c047-0f5c-4378-bcd8-9cad491450b7" />

efuenzalida52 committed:
removed unneeded/extra div

taija committed:
Merge branch 'release-v1.5' into feat-differentiator-anchor

taija commented:
@efuenzalida52  - FYI, I just fixed some merge conflicts with the release branch, so if you need to make any further changes to this branch make sure you pull first.

taija reviewed (approved):
Code review looks good, surface level testing also looks good!

taija reviewed (changes requested):
Approval revoked! Something in this PR broke 'icon' based differentiators, but I'm not sure what. 
<img width="1229" height="564" alt="Screenshot 2025-10-22 at 4 04 22 PM" src="https://github.com/user-attachments/assets/6ca4f494-3c51-4665-a039-523134344dd7" />

taija committed:
🐛 FIX: Diff Icon Text and Spacing

taija reviewed (approved):
Fixed!

TaylorYeeBC reviewed (approved):
Code reviewed and tested on Staging site. Icons are where they're supposed to be and tagging works! 🎊
